presto-orc/src/test/java/com/facebook/presto/orc/TestSelectiveOrcReader.java

```diff
@@ -555,6 +568,152 @@ public void testDecimalSequence()
                                 false, true, 28999999999L, 28999999999L, false, true, true))));
     }
 
+    @Test
+    public void testSliceVarchar()
+            throws Exception
+    {
+        //dictionary
+        tester.testRoundTrip(VARCHAR, newArrayList(limit(cycle(ImmutableList.of("apple", "apple pie", "apple\uD835\uDC03", "apple\uFFFD")), 3000)),
```

Comment:
- any reason to use 3000 and not NUM_ROWS? 3000 rows will make only 1 row group.
- consider using NUM_ROWS throughout
- consider adding helper methods to make filters; this could make the code easier to read

```
        // dictionary
        tester.testRoundTrip(VARCHAR, newArrayList(limit(cycle(ImmutableList.of("apple", "apple pie", "apple\uD835\uDC03", "apple\uFFFD")), NUM_ROWS)),
                stringEquals(false, "apple"));

        // direct
        tester.testRoundTrip(VARCHAR,
                intsBetween(0, NUM_ROWS).stream().map(Object::toString).collect(toList()),
                stringIn(false, "10", "11"),
                stringIn(true, "10", "11"),
                stringBetween(false, "14", "10"));

        // direct & dictionary with filters
        tester.testRoundTripTypes(
                ImmutableList.of(VARCHAR, VARCHAR),
                ImmutableList.of(
                        intsBetween(0, NUM_ROWS).stream().map(Object::toString).collect(toList()),
                        newArrayList(limit(cycle(ImmutableList.of("A", "B", "C")), NUM_ROWS))),
                toSubfieldFilters(ImmutableMap.of(
                        0, stringBetween(true, "16", "10"),
                        1, stringBetween(false, "B", "A"))));


    private static TupleDomainFilter stringBetween(boolean nullAllowed, String upper, String lower)
    {
        return BytesRange.of(lower.getBytes(), false, upper.getBytes(), false, nullAllowed);
    }

    private static TupleDomainFilter stringEquals(boolean nullAllowed, String value)
    {
        return BytesRange.of(value.getBytes(), false, value.getBytes(), false, nullAllowed);
    }

    private static TupleDomainFilter stringIn(boolean nullAllowed, String... values)
    {
        return BytesValues.of(Arrays.stream(values).map(String::getBytes).toArray(byte[][]::new), nullAllowed);
    }
```